consensus/src/lib.rs

```diff
@@ -208,70 +245,39 @@ impl<N: GossipNetwork, S: Storage> Consensus<N, S> {
         known_peers: &[Peer],
         timestamp: Timestamp,
     ) -> Result<Vec<ProgressResult>, Error> {
-        let messages = self.dms.read_messages().await?;
-        let messages = messages
+        self.abort_if_finalized()?;
+        let messages = self
+            .dms
+            .read_messages()
+            .await?
             .into_iter()
             .filter(|m| !self.state.updated_messages.contains(&m.to_hash256()))
             .collect::<Vec<_>>();
-        let mut final_result = Vec::new();
-        for message in messages {
-            let consensus_message = serde_json::from_str::<ConsensusMessage>(message.data())
-                .expect("this must be already verified by the message filter");
+        // Not directly commit the change, by saving a copy of the vetomint FSM, in case of possible DMS errors.
```

Comment:
```suggestion
        // Not directly commit the change, save a copy of the vetomint FSM due to possible DMS errors.
```

Reply:
Will be squashed into the next commit.

Reply:
Updated the comment in a clearer way. please check.